core/java/iesi-core/src/main/java/io/metadew/iesi/connection/database/bigquery/BigqueryDatabaseService.java

```diff
@@ -0,0 +1,256 @@
+        connectionUrl.append(";OAuthServiceAcctEmail=");
+        connectionUrl.append(serviceAccount);
+        connectionUrl.append(";OAuthPvtKeyPath=");
+        connectionUrl.append(keyPath);
+        connectionUrl.append(";");
+
+        return connectionUrl.toString();
+    }
+
+    private static String getConnectionUrlUserAccount(String hostName, int portNumber, String project, int authValue) {
+        StringBuilder connectionUrl = new StringBuilder();
+        connectionUrl.append("jdbc:bigquery://");
+        connectionUrl.append(hostName);
+        if (portNumber > 0) {
+            connectionUrl.append(":");
+            connectionUrl.append(portNumber);
+        }
+        connectionUrl.append(";ProjectId=");
+        connectionUrl.append(project);
+        connectionUrl.append(";OAuthType=");
+        connectionUrl.append(authValue);
+        connectionUrl.append(";");
+
+        return connectionUrl.toString();
+    }
+
+    private static String getConnectionUrlWithTokens(String hostName, int portNumber, String project, int authValue, String accessToken, String refreshToken, String clientId, String clientSecret) {
+        StringBuilder connectionUrl = new StringBuilder();
+        connectionUrl.append("jdbc:bigquery://");
+        connectionUrl.append(hostName);
+        if (portNumber > 0) {
+            connectionUrl.append(":");
+            connectionUrl.append(portNumber);
+        }
+        connectionUrl.append(";ProjectId=");
+        connectionUrl.append(project);
+        connectionUrl.append(";OAuthType=");
+        connectionUrl.append(authValue);
+        if (!accessToken.isEmpty()) {
+            connectionUrl.append(";OAuthAccessToken=");
+            connectionUrl.append(accessToken);
+        }
+        if (!refreshToken.isEmpty()) {
+            connectionUrl.append(";OAuthRefreshToken=");
+            connectionUrl.append(refreshToken);
+        }
+        if (!clientId.isEmpty()) {
+            connectionUrl.append(";OAuthClientId=");
+            connectionUrl.append(clientId);
+        }
+        if (!clientSecret.isEmpty()) {
+            connectionUrl.append(";OAuthClientSecret=");
+            connectionUrl.append(clientSecret);
+        }
+        connectionUrl.append(";");
+
+        return connectionUrl.toString();
+    }
+
+    private static String getConnectionUrlDefaultCredentials(String hostName, int portNumber, String project, int authValue) {
+        StringBuilder connectionUrl = new StringBuilder();
+        connectionUrl.append("jdbc:bigquery://");
+        connectionUrl.append(hostName);
+        if (portNumber > 0) {
+            connectionUrl.append(":");
+            connectionUrl.append(portNumber);
+        }
+        connectionUrl.append(";ProjectId=");
+        connectionUrl.append(project);
+        connectionUrl.append(";OAuthType=");
+        connectionUrl.append(authValue);
+        connectionUrl.append(";");
+
+        return connectionUrl.toString();
+    }
```

Comment:
Should be deleted/moved once subclasses of 'BigqueryDatabaseConnection' are implemented

Reply:
Resolve in next commit